MediaBrowser.Server.Mono/Program.cs

```diff
@@ -83,16 +85,43 @@ private static void SetSqliteProvider()
             SQLitePCL.Batteries_V2.Init();
         }
 
-        private static ServerApplicationPaths CreateApplicationPaths(string applicationPath, string programDataPath)
+        private static ServerApplicationPaths CreateApplicationPaths(
+            string applicationPath,
+            string programDataPath)
         {
-            if (string.IsNullOrEmpty(programDataPath))
+            var appFolderPath = Path.GetDirectoryName(applicationPath);
+
+            if (!string.IsNullOrEmpty(programDataPath))
             {
-                programDataPath = ApplicationPathHelper.GetProgramDataPath(applicationPath);
+                return new ServerApplicationPaths(
+                    programDataPath,
+                    appFolderPath,
+                    Path.GetDirectoryName(applicationPath));
             }
 
-            var appFolderPath = Path.GetDirectoryName(applicationPath);
-
-            return new ServerApplicationPaths(programDataPath, appFolderPath, Path.GetDirectoryName(applicationPath));
+            var useDebugPaths = false;
+#if DEBUG
+            useDebugPaths = true;
+#endif
+            
+            var debugProgramDataPath = ConfigurationManager.AppSettings["DebugProgramDataPath"];
```

Comment:
Moved calls to `ConfigurationManager` outside of the `ApplicationPathHelper` class, otherwise the class isn't testable, as calls to `ConfigurationManager` don't contain any configuration when unit testing. Instead, these configuration values are passed into the `ApplicationPathHelper` through its constructor. This means appropriate values can be injected when testing.